apps/examples/testcase/le_tc/kernel/tc_libc_pthread.c

```diff
@@ -1227,6 +1230,85 @@ static void tc_libc_pthread_pthread_rwlock_trywrlock(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @fn                   :tc_libc_pthread_pthread_setcancelstate
+* @brief                :This tc tests pthread_setcancelstate()
+* @Scenario             :The function shall atomically both set the calling thread's cancelability state to the indicated state 
+*                        and return the previous cancelability state at the location referenced by oldstate
+*                        If successful pthread_setcancelstate() function shall return zero;
+*                        otherwise, an error number shall be returned to indicate the error.
+* @API'scovered         :pthread_setcancelstate
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void tc_libc_pthread_pthread_setcancelstate(void)
```

Comment:
As discussed on chat, 
When CONFIG_CANCELLATION_POINTS is enabled 
os/kernel/task/ : task_setcanceltype.c & task_testcancel.c will be compiled
When CONFIG_CANCELLATION_POINTS is disabled 
lib/libc/sched/ : task_setcanceltype.c & task_testcancel.c will be compiled

Now, pthread_setcanceltype, pthread_setcancelstate , pthread_testcancel  use above mentioned task API's 

Hence, either we don't include any flag at all 
OR
We add 
under tc_libc_pthread.c >>
#ifndef CONFIG_CANCELLATION_POINTS 
        tc_libc_pthread_pthread_setcanceltype();
        tc_libc_pthread_pthread_testcancel();
#endif

under tc_pthread.c >>
#ifdef CONFIG_CANCELLATION_POINTS 
        tc_libc_pthread_pthread_setcanceltype();
        tc_libc_pthread_pthread_testcancel();
#endif